### PR TITLE
Add Bar Chart for Traffic Deaths by Year

### PR DESCRIPTION
**Pull request by iuddin**
### What changes did you make and why did you make them ?

- Integrated Chart.js and react-chartjs-2 (React integration)
- Rendered a sample bar chart with sample data
- Set up server-side proxy route at '/api/who-data' to fetch data from remote API (WHO's API), enabling CORS compliance and encapsulating API request logic
- Prepared fetched traffic deaths data for bar chart rendering

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,16 +1,98 @@
-import Image from 'next/image'
+'use client'
+
+import React  from 'react';
+import { Bar, Line } from 'react-chartjs-2';
+import Image from 'next/image';
+import useProcessedData from './panel-one/page';
+import { faker } from '@faker-js/faker';
```
Comment (farskid) [inline src/app/page.tsx:7]: Let's remove this.

File: src/app/data/AbsDeaths.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export default function useAbsDeaths() {
+  const [absDeathsData, setAbsDeathsData] = useState(null);
+
+  useEffect(() => {
+    const fetchAbsDeaths = async () => {
+      const res = await fetch('/api/who-data');
+      const jsonData = await res.json();
+      // console.log(jsonData.data.value[0]);
+      setAbsDeathsData(jsonData.data);
+    };
```
Comment (farskid) [inline src/app/data/AbsDeaths.tsx]: This can be hoisted to outside `useEffect`.

File: src/app/panel-one/page.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const useProcessedData = () => {
+    const absDeathsData = useAbsDeaths();
+    // console.log('absDeathsData Arr', absDeathsData);
+
+    const [globalDeathsByYear, setglobalDeathsByYear] = useState({})
+
+    useEffect(() => {
+        if (absDeathsData) {
+            let totalDeaths2019: number = Math.round(getTotalsByYear(absDeathsData, 2019, 'mean'));
```
Comment (farskid) [inline src/app/panel-one/page.tsx]:
```suggestion
            let totalDeaths2019 = Math.round(getTotalsByYear(absDeathsData, 2019, 'mean'));
```

File: src/app/panel-one/page.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        }
+    }, [absDeathsData])
+
+    return globalDeathsByYear;
+}
+
+
+//create total # deaths / year object
+function getTotalsByYear (array: any, year: number, quartile: string) {
```
Comment (farskid) [inline src/app/panel-one/page.tsx]: Let's mark the parameter and return type for this functions.

**Review by farskid — APPROVED** [review]

**Comment by farskid** [comment]
@iuddin Approved this PR. Aside from those nit comments, this PR looks good to me. Let's address them and merge.

File: src/app/page.tsx (hunk 1 above)
Comment (iuddin) [reply src/app/page.tsx:7]: Done.

File: src/app/data/AbsDeaths.tsx (hunk 2 above)
Comment (iuddin) [reply src/app/data/AbsDeaths.tsx]: Done.

File: src/app/panel-one/page.tsx (hunk 3 above)
Comment (iuddin) [reply src/app/panel-one/page.tsx]: Done.

File: src/app/panel-one/page.tsx (hunk 4 above)
Comment (iuddin) [reply src/app/panel-one/page.tsx]: Updated the getTotalsByYear function with explicit types and added comments for clarity.

**Pull request merged**